### PR TITLE
MULE-12663: In Message Filter, When unaccepted events are handled, execution of flow caller is not stopped.

### PR DESCRIPTION
**Pull request by facundovs**
In a subflow with a filter, if the filter condition is not satisfied and the unaccepted event is handled, then the flow that called this subflow must stop its execution.

### CODE REVIEW

File: transports/rmi/src/test/java/org/mule/transport/issues/RmiMethodTypeMule1857TestCase.java [hunk 1]
```diff
@@ -31,3 +31,4 @@ public RmiMethodTypeMule1857TestCase(ConfigVariant variant, String configResourc
     }      
     
 }
+//
```
Comment (dfeist) [inline transports/rmi/src/test/java/org/mule/transport/issues/RmiMethodTypeMule1857TestCase.java]: revert

File: transports/rmi/src/test/resources/jnp-functional-test-flow.xml [hunk 2]
```diff
@@ -43,6 +43,7 @@
     <sub-flow name="Sender2SubFlow">
         <payload-type-filter expectedType="java.lang.String"/>
         <outbound-endpoint ref="Sender2"/>
+        <component class="org.mule.transport.AbstractFunctionalTestCase$TestJavaComponent"/>
```
Comment (dfeist) [inline transports/rmi/src/test/resources/jnp-functional-test-flow.xml]: Is this for same issue, it wasn't in 3.8 PR

File: transports/rmi/src/test/java/org/mule/transport/AbstractFunctionalTestCase.java [hunk 3]
```diff
@@ -46,13 +52,28 @@ public void testCase() throws Exception
 
         // send String
         message = client.send("vm://testin", "test matching component first time", null);
-        assertNotNull(message);
-        assertEquals(message.getPayload(), "emit tsrif tnenopmoc gnihctam tset");
+        if(variant.equals(SERVICE))
```
Comment (dfeist) [inline transports/rmi/src/test/java/org/mule/transport/AbstractFunctionalTestCase.java]: Is this for same issue, it wasn't in 3.8 PR

Comment (facundovs) [inline transports/rmi/src/test/java/org/mule/transport/AbstractFunctionalTestCase.java]: Yes It is. 
As in 3.x the default behavior changes, I had to modify some tests.

File: transports/rmi/src/test/resources/jnp-functional-test-flow.xml (hunk 2 above)
Comment (facundovs) [inline transports/rmi/src/test/resources/jnp-functional-test-flow.xml]: Yes It is.
As in 3.x the default behavior changes, I had to modify some tests.

**Pull request merged**